src/Aquifer.Common/Utilities/VersificationUtilities.cs

```diff
@@ -1,19 +1,258 @@
+        var endVerse = BibleUtilities.TranslateVerseId(endVerseId);
+
+        var maxChapterNumberAndVerseNumbersByBookIdMap =
+            await versificationService.GetMaxChapterNumberAndVerseNumbersByBookIdMapAsync(bibleId, ct);
+
+        var excludedVerseIds = await versificationService.GetExcludedVerseIdsAsync(bibleId, ct);
+
+        var expandedVerseIds = new List<int>();
+        for (var bookId = startVerse.bookId; bookId <= endVerse.bookId; bookId++)
+        {
+            // Bibles often don't include every possible book.
+            if (!maxChapterNumberAndVerseNumbersByBookIdMap.TryGetValue(bookId, out var maxChapterNumberAndVerseNumbers))
+            {
+                continue;
+            }
+
+            var startChapterNumber = bookId == startVerse.bookId
+                ? startVerse.chapter
+                : 1;
+            var endChapterNumber = bookId == endVerse.bookId
+                ? endVerse.chapter
+                : maxChapterNumberAndVerseNumbers.MaxChapterNumber;
+
+            for (var chapterNumber = startChapterNumber; chapterNumber <= endChapterNumber; chapterNumber++)
+            {
+                // Edge Case: It's theoretically possible that a Bible could be missing an entire chapter.
+                if (!maxChapterNumberAndVerseNumbers.MaxVerseNumberByChapterNumberMap
+                        .TryGetValue(chapterNumber, out var maxVerseNumber))
+                {
+                    continue;
+                }
+
+                var startVerseNumber = bookId == startVerse.bookId && chapterNumber == startVerse.chapter
+                    ? startVerse.verse
+                    : 1;
+                var endVerseNumber = bookId == endVerse.bookId && chapterNumber == endVerse.chapter
+                    ? endVerse.verse
+                    : maxVerseNumber;
+
+                for (var verseNumber = startVerseNumber; verseNumber <= endVerseNumber; verseNumber++)
+                {
+                    var verseId = BibleUtilities.GetVerseId(bookId, chapterNumber, verseNumber);
+
+                    // Some Bibles are missing specific verses.
+                    if (!excludedVerseIds.Contains(verseId))
+                    {
+                        expandedVerseIds.Add(verseId);
+                    }
+                }
+            }
+        }
+
+        return expandedVerseIds;
+    }
+
+    /// <summary>
+    /// Returns the verse ID in the target Bible that corresponds to the given verse ID in the source Bible.
+    /// <c>null</c> will be returned if it's not possible to map between the versification schemes.
+    /// </summary>
+    public static async Task<int?> ConvertVersificationAsync(
+        int sourceBibleId,
+        int sourceVerseId,
+        int targetBibleId,
+        ICachingVersificationService versificationService,
+        CancellationToken ct)
+    {
+        if (!await IsValidVerseIdAsync(sourceBibleId, sourceVerseId, versificationService, ct))
```

Comment:
I decided to have these utilities throw an exception if the provided source verse IDs are not in the given Bible.  We could change this if desired but it seems to me that any caller should already be doing the right thing regarding verse IDs.

We'll also have the `VersificationUtilities.IsValidVerseIdAsync()` method available to any endpoints that want to validate  API consumer input.